ehrql/tables/beta/tpp.py

```diff
@@ -42,23 +42,104 @@
 
 @table
 class addresses(EventFrame):
-    address_id = Series(int)
-    start_date = Series(datetime.date)
-    end_date = Series(datetime.date)
-    address_type = Series(int)
-    rural_urban_classification = Series(int)
-    imd_rounded = Series(int, constraints=[Constraint.ClosedRange(0, 32_800, 100)])
+    """
+    Addresses
+
+    Geographic characteristics of the home address a patient registers with a practice.
+    Each row in this table is one registration period per patient.
+    Occasionally, a patient has multiple active registrations on a given date.
+    The postcode from the address is mapped to an Output Area,
+    from which other larger geographic representations can be derived
+    (see various [ONS publications][addresses_ukgeographies] for more detail).
+
+    !!! tip
+        Avoid extracting the rounded IMD ranking to a binary format, such as `.feather` or `.dta`.
+        Either nest it within a variable, such as when grouping rounded IMD by quintile,
+        or extract it to a non-binary format, such as .csv.gz.
+
+        To group rounded IMD ranks by quintile:
+
+        ```py
+        imd = addresses.for_patient_on("2023-01-01").imd_rounded
+        dataset.imd_quintile = case(
+            when((imd >=0) & (imd < int(32844 * 1 / 5))).then("1 (most deprived)"),
+            when(imd < int(32844 * 2 / 5)).then("2"),
+            when(imd < int(32844 * 3 / 5)).then("3"),
+            when(imd < int(32844 * 4 / 5)).then("4"),
+            when(imd < int(32844 * 5 / 5)).then("5 (least deprived)"),
+            default="unknown"
+        )
+        ```
+
+    [addresses_ukgeographies]: https://www.ons.gov.uk/methodology/geography/ukgeographies
+    """
+
+    address_id = Series(
+        int,
+        description="Unique address identifier.",
+    )
+    start_date = Series(
+        datetime.date,
+        description="Start date of registration period.",
+    )
+    end_date = Series(
+        datetime.date,
+        description="End date of registration period.",
+    )
+    address_type = Series(
+        int,
+        description="Type of address.",
+    )
+    rural_urban_classification = Series(
+        int,
+        description="""
+            Rural urban classification:
+
+            * 1 - Urban major conurbation
+            * 2 - Urban minor conurbation
+            * 3 - Urban city and town
+            * 4 - Urban city and town in a sparse setting
+            * 5 - Rural town and fringe
+            * 6 - Rural town and fringe in a sparse setting
+            * 7 - Rural village and dispersed
+            * 8 - Rural village and dispersed in a sparse setting
+        """,
+        constraints=[Constraint.ClosedRange(1, 8, 1)],
```

Comment:
I added this `Constraint.ClosedRange(1, 8, 1)` because it seems logical from the description we use in our current docs, but I didn't check what the data looks like.

Reply:
micro-nit: You could do `Constraint.ClosedRange(1, 8)`, because the `steps` parameter [defaults to 1](https://github.com/opensafely-core/ehrql/blob/4fdb6452aa1333310e1f0530ff66ee5788426706/ehrql/query_model/table_schema.py#L47).